monash-human-power/dashboard
# Amytjea/crash on dash

## Description
Adds a pop up modal alerting to the crash of V3 to all pages which monitor V3 (i.e Dashboard, Logs, Status, Boost and Camera system pages when observing V3)

The modal will appear when the crash detection topic (/v3/wireless_module/3/crash_detection) sends a true value.
It will provide the timestamp of the latest detected crash, along with the number of crashes which have occurred if the modal hasn't been dismissed and the bike has crashed again.


## Screenshots
![image](https://github.com/monash-human-power/dashboard/assets/49554284/8871b700-0fb8-4bed-a71b-3645c31f59f7)

![image](https://github.com/monash-human-power/dashboard/assets/49554284/bf2a8d32-b158-48f4-a2ed-6e61ccd8ff5e)



## Steps to Test
Use the following command  in the terminal to simulate a crash occurring:

```mosquitto_pub -t "/v3/wireless_module/3/crash_detection" -m '{\"value\":true}'```

To simulate a false value, simply change the above to false.

Use the command multiple times to check the counter works as intended
